### PR TITLE
add testing module

### PR DESCRIPTION
**Pull request by itajaja**
it was worthwile to change the tests to dogfood the testing module

### CODE REVIEW

File: flask_resty/testing.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        assert actual == expected
+
+
+def assert_response(response, expected_status=200, expected_data=NO_DATA):
+    """check the results of a response. The data is checked against either the
+    data or the errors in the body, depending on the expected status. It is
+    allowed for the data to have more keys than the one specified"""
+    assert response.status_code == expected_status
+
+    if expected_data == NO_DATA:
```
Comment (taion) [inline flask_resty/testing.py]: you should probably make `NO_DATA = object()`

File: tests/test_auth.py [hunk 2]
```diff
@@ -3,11 +3,12 @@
     Api, ApiError, AuthenticationBase, GenericModelView,
     HasAnyCredentialsAuthorization, HasCredentialsAuthorizationBase,
 )
+from flask_resty.testing import assert_response
 from marshmallow import fields, Schema
 import pytest
 from sqlalchemy import Column, Integer, String
 
-import helpers
+from helpers import request
```
Comment (taion) [inline tests/test_auth.py]: did this really need to change? ಠ_ಠ

File: flask_resty/testing.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    elif isinstance(expected, list):
+        for a, e in zip_longest(actual, expected):
+            assert_value(a, e)
+    elif isinstance(expected, float):
+        assert abs(actual / expected - 1) < 1e-6
+    else:
+        assert actual == expected
+
+
+def assert_response(response, expected_status=200, expected_data=NO_DATA):
```
Comment (taion) [inline flask_resty/testing.py]: i'd make `expected_status` mandatory

Comment (itajaja) [inline flask_resty/testing.py]: changed it already

File: tests/test_auth.py (hunk 2 above)
Comment (itajaja) [inline tests/test_auth.py]: meh

File: flask_resty/testing.py (hunk 3 above)
Comment (itajaja) [inline flask_resty/testing.py]: mkay

Comment (taion) [inline flask_resty/testing.py]: i already did it btw

Comment (itajaja) [inline flask_resty/testing.py]: yeah I saw that :P

**Pull request merged**